src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -390,140 +255,360 @@ await _logger.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1803,
                 {
                     _success = false;
                 }
-                telemetry.EndIntervalMeasure(ValidateRestoreGraphsDuration);
 
+                telemetry.EndIntervalMeasure(ValidateRestoreGraphsDuration);
 
                 // Generate Targets/Props files
                 var msbuildOutputFiles = Enumerable.Empty<MSBuildOutputFile>();
                 string assetsFilePath = null;
                 string cacheFilePath = null;
 
-                using (telemetry.StartIndependentInterval(CreateRestoreResultDuration))
+                (msbuildOutputFiles,
+                    assetsFilePath,
+                    cacheFilePath,
+                    assetsFile,
+                    graphs,
+                    packagesLockFile,
+                    packagesLockFilePath,
+                    cacheFile) = await ProcessRestoreResultAsync(
+                    telemetry,
+                    localRepositories,
+                    contextForProject,
+                    isLockFileValid,
+                    regenerateLockFile,
+                    msbuildOutputFiles,
+                    assetsFilePath,
```

Comment:
This is directly assigned, so it doesn't need to be passed in. It's null everytime anyways.

Reply:
All 3 defined variables have the same issue.